src/en/hentairead/src/eu/kanade/tachiyomi/extension/en/hentairead/Hentairead.kt

```diff
@@ -189,14 +189,18 @@ class Hentairead : Madara("HentaiRead", "https://hentairead.com", "en", dateForm
     override fun pageListParse(document: Document): List<Page> {
         launchIO { countViews(document) }
 
-        val pages = document.selectFirst("[id=single-chapter-js-extra]")?.data()
-            ?.substringAfter(":[")
-            ?.substringBefore("],")
-            ?.let { json.decodeFromString<List<PageDto>>("[$it]") }
+        val pageBaseUrl = document.selectFirst("[id=single-chapter-js-extra]")?.data()
+            ?.substringAfter("chapterExtraData = ")
+            ?.substringBefore(";")?.let { json.decodeFromString<ImageBaseUrlDto>(it).baseUrl }
+
+        val pages = document.selectFirst("[id=single-chapter-js-before]")?.data()
+            ?.substringAfter("= '")
+            ?.substringBefore("'")
+            ?.let { json.decodeFromString<PagesDto>(String(Base64.decode(it, Base64.DEFAULT))) }
             ?: throw Exception("Failed to find page list. Non-English entries are not supported.")
 
-        return pages.mapIndexed { idx, page ->
-            Page(idx, document.location(), page.src)
+        return pages.data.chapter.images.mapIndexed { idx, page ->
+            Page(idx, document.location(), "$pageBaseUrl/${page.src}")
```

Comment:
Does this escape URLs properly?

Assuming the least, it would break in this scenario:
- `page.src` is a filename
- the filename is `example#filename.png`
- it will try to load `$pageBaseUrl/example` instead of `$pageBaseUrl/example%23filename.png`

Reply:
I think its fine the pages are saved like this 
```json
[{"src":"239236\/66227\/hr_0001.jpg","width":1280,"height":1810}...n]
```
and the baseUrl like this
```json
{ "baseUrl": "https:\/\/henread.xyz" }
```

Reply:
Thanks for clarifying. If this was escaped, it would turn `/` into `%2F`, which would be wrong.